src/neo/Cryptography/Crypto.cs

```diff
@@ -59,21 +62,50 @@ public static byte[] Sign(byte[] message, byte[] prikey, byte[] pubkey)
         /// <returns><see langword="true"/> if the signature is valid; otherwise, <see langword="false"/>.</returns>
         public static bool VerifySignature(ReadOnlySpan<byte> message, ReadOnlySpan<byte> signature, ECC.ECPoint pubkey)
         {
-            ECCurve curve =
-                pubkey.Curve == ECC.ECCurve.Secp256r1 ? ECCurve.NamedCurves.nistP256 :
-                pubkey.Curve == ECC.ECCurve.Secp256k1 ? ECCurve.CreateFromFriendlyName("secP256k1") :
-                throw new NotSupportedException();
-            byte[] buffer = pubkey.EncodePoint(false);
-            using var ecdsa = ECDsa.Create(new ECParameters
+            if (signature.Length != 64) return false;
+
+            if (IsOSX && pubkey.Curve != ECC.ECCurve.Secp256r1)
             {
-                Curve = curve,
-                Q = new ECPoint
+                try
                 {
-                    X = buffer[1..33],
-                    Y = buffer[33..]
+                    var curve = Org.BouncyCastle.Asn1.Sec.SecNamedCurves.GetByName("secp256k1");
+                    var domain = new Org.BouncyCastle.Crypto.Parameters.ECDomainParameters(curve.Curve, curve.G, curve.N, curve.H);
+                    var point = curve.Curve.CreatePoint(
+                        new Org.BouncyCastle.Math.BigInteger(pubkey.X.Value.ToString()),
+                        new Org.BouncyCastle.Math.BigInteger(pubkey.Y.Value.ToString()));
+                    var pubKey = new Org.BouncyCastle.Crypto.Parameters.ECPublicKeyParameters("ECDSA", point, domain);
+                    var signer = Org.BouncyCastle.Security.SignerUtilities.GetSigner("SHA-256withECDSA");
+
+                    signer.Init(false, pubKey);
+                    signer.BlockUpdate(message.ToArray(), 0, message.Length);
+
+                    var sig = signature.ToArray();
+                    var r = new Org.BouncyCastle.Math.BigInteger(1, sig, 0, 32);
+                    var s = new Org.BouncyCastle.Math.BigInteger(1, sig, 32, 32);
+                    sig = new Org.BouncyCastle.Asn1.DerSequence(new Org.BouncyCastle.Asn1.DerInteger(r), new Org.BouncyCastle.Asn1.DerInteger(s)).GetDerEncoded();
+
+                    return signer.VerifySignature(sig);
                 }
-            });
-            return ecdsa.VerifyData(message, signature, HashAlgorithmName.SHA256);
+                catch { return false; }
```

Comment:
Why do we need this `try-catch`?

Reply:
@ZhangTao1596 could you test it without the try block in osx?

Reply:
I would suggest keeping `try-catch` as you are using a third-party lib. 
Especially 
`var point = curve.Curve.CreatePoint(
                        new Org.BouncyCastle.Math.BigInteger(pubkey.X.Value.ToString()),
                        new Org.BouncyCastle.Math.BigInteger(pubkey.Y.Value.ToString()));`
 might throw exception if `pubkey` is not a valid point on `secp256k1`.

Reply:
> @ZhangTao1596 could you test it without the try block in osx?

I think it's OK since we already went into 'try', but we can't go to 'catch'. After remove, no effect on the result.

Reply:
https://github.com/bcgit/bc-csharp/blob/93b32a75656955faf7996d0f3e0ed391968d2ac6/crypto/src/crypto/signers/ECDsaSigner.cs#L70

Reply:
Well, I still suggest keeping the `try-catch` for a **THIRD-PARTY** lib, unless you know it very well.